lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -4567,3 +4606,139 @@ void send_s11_modify_bearer_request(
+  memcpy(
+      s11_delete_bearer_command->ebi_list.ebis, pdn_context->ebi_to_be_del,
+      sizeof(s11_delete_bearer_command->ebi_list.ebis));
+
+  OAILOG_INFO_UE(
+      LOG_MME_APP, ue_mm_context->emm_context._imsi64,
+      "Sending S11_DELETE_BEARER_COMMAND to SPGW task for " TEID_FMT "\n",
+      pdn_context->s_gw_teid_s11_s4);
+  send_msg_to_task(&mme_app_task_zmq_ctx, TASK_SPGW, message_p);
+  ue_mm_context->nb_delete_bearer_cmd++;
+  OAILOG_FUNC_OUT(LOG_MME_APP);
+}
+
+// Deletes the bearer context
+void mme_app_handle_mme_init_local_deactivation(
+    mme_app_desc_t* mme_app_desc_p,
+    itti_s11_nw_init_deactv_bearer_request_t* const bearer_deactv_req_p) {
+  OAILOG_FUNC_IN(LOG_MME_APP);
+  ue_mm_context_t* ue_context_p = NULL;
+  ebi_t ebi                     = 0;
+
+  ue_context_p = mme_ue_context_exists_s11_teid(
+      &mme_app_desc_p->mme_ue_contexts, bearer_deactv_req_p->s11_mme_teid);
+
+  if (ue_context_p == NULL) {
+    OAILOG_ERROR(
+        LOG_MME_APP,
+        "Cannot fetch UE context for teid: " TEID_FMT
+        " received in bearer_deactv_req_p from spgw\n",
+        bearer_deactv_req_p->s11_mme_teid);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+  if (!bearer_deactv_req_p) {
+    OAILOG_ERROR_UE(
+        LOG_MME_APP, ue_context_p->emm_context._imsi64,
+        "bearer_deactv_req_p received from spgw is NULL\n");
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+  OAILOG_DEBUG(
+      LOG_MME_APP,
+      "Received bearer_deactv_req_p from SGW for S11 teid" TEID_FMT
+      "," MME_UE_S1AP_ID_FMT "\n",
+      bearer_deactv_req_p->s11_mme_teid, ue_context_p->mme_ue_s1ap_id);
+
+  /* Fetch PDN context using any of the bearer ids as all the bearers
+   * belong to the same PDN
+   */
+  pdn_cid_t pid =
+      ue_context_p->bearer_contexts[EBI_TO_INDEX(bearer_deactv_req_p->ebi[0])]
+          ->pdn_cx_id;
+  if (pid >= MAX_APN_PER_UE) {
+    OAILOG_ERROR_UE(
+        ue_context_p->emm_context._imsi64, LOG_NAS_ESM,
+        "No PDN connection found for pid=%d, EBI=%d while processing \n", pid,
+        bearer_deactv_req_p->ebi[0]);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  pdn_context_t* pdn_context = ue_context_p->pdn_contexts[pid];
+  if (!pdn_context) {
+    OAILOG_ERROR_UE(
+        ue_context_p->emm_context._imsi64, LOG_NAS_ESM,
+        "PDN context is NULL for pid=%d for UE: " MME_UE_S1AP_ID_FMT "\n", pid,
+        ue_context_p->mme_ue_s1ap_id);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+  // Delete Eps bearer context
+  for (uint8_t itr = 0; itr < bearer_deactv_req_p->no_of_bearers; itr++) {
+    ebi = bearer_deactv_req_p->ebi[itr];
+    bearer_context_t* bearer_context =
+        mme_app_get_bearer_context(ue_context_p, ebi);
+    if (bearer_context) {
+      int bearer_idx = EBI_TO_INDEX(ebi);
+      eps_bearer_release(&ue_context_p->emm_context, ebi, &pid, &bearer_idx);
+      // Remove dedicated bearer context
+      free_wrapper((void**) &ue_context_p->bearer_contexts[bearer_idx]);
```

Comment:
isn't `bearer_contexts` an array? how is this not crashing?

Reply:
its an array of pointers

Reply:
I see an array of "bearer_context_within_create_bearer_response_t". What am I missing?

Reply:
Here I am freeing the bearer context which is within the ue_context_p. Its an array of pointers which is allocated during bearer creation in mme_app_create_bearer_context(). This needs to be freed during bearer release. 
![image](https://user-images.githubusercontent.com/38202471/143999848-f6b00639-c028-4184-baf6-f22a9c2df326.png)